### DISCUSSION TITLE
When I try to run the code in Stitching Tutorial.ipynb, there is an error happening in the part of "Crop".

### DISCUSSION THREAD

**Discussion by FuSungLin** [OpenStitching/lir#28] [Q&A]
Hello, 
I'm trying to run the code:
`lir = cropper.estimate_largest_interior_rectangle(mask)`
However, I encountered an error:
**------------------------**
**TypingError: Failed in nopython mode pipeline (step: nopython frontend)**
Cannot unify array(uint32, 2d, A) and array(uint32, 1d, A) for 's', defined at C:\Users\user\anaconda3\lib\site-packages\largestinteriorrectangle\lir_basis.py (99)

File "C:\Users\user\anaconda3\lib\site-packages\largestinteriorrectangle\lir_basis.py", line 99:
def span_map(grid, h_adjacency, v_adjacency):
    <source elided>
        v_vec = v_vector(v_adjacency, x, y)
        s = spans(h_vec, v_vec)
        ^

[1] During: typing of assignment at C:\Users\user\anaconda3\lib\site-packages\largestinteriorrectangle\lir_basis.py (100)

File "C:\Users\user\anaconda3\lib\site-packages\largestinteriorrectangle\lir_basis.py", line 100:
def span_map(grid, h_adjacency, v_adjacency):
    <source elided>
        s = spans(h_vec, v_vec)
        s = biggest_span(s)
        ^
**------------------------**

Also, when I try to give another valuable name, like 's0', to replace the front 's' in `s = biggest_span(s)`, it still shows the error:
**------------------------**
**TypeError: Failed in nopython mode pipeline (step: convert to parfors)**
'NoneType' object is not iterable
**------------------------**

I don't have any idea about this so far. Could you giva me some help?

**Comment by lukasalexanderweber** [comment]
Have you checked https://stackoverflow.com/questions/55065419/typingerror-failed-in-nopython-mode-pipeline-step-nopython-frontend?

**Reply by lukasalexanderweber** [reply]
Maybe somethings wrong with `mask`

**Reply by FuSungLin** [reply]
Hi, 
I have checked the web you gave, but still not found the solution.
I have saved the entire Jupyter Notebook message as the attached PDF below:
[Stitching Tutorial - Jupyter Notebook.pdf](https://github.com/OpenStitching/lir/files/13807960/Stitching.Tutorial.-.Jupyter.Notebook.pdf)

Addintionally, I found a solution by turning off the parallel setting in line 90 of lir_basis.py:
`@nb.njit('uint32[:, :, :](boolean[:,::1], uint32[:,::1], uint32[:,::1])',parallel=False, cache=True)`
However, another error occurred, as shown in the attached PDF below:
[Stitching Tutorial - Jupyter Notebook - parallel=False.pdf](https://github.com/OpenStitching/lir/files/13808488/Stitching.Tutorial.-.Jupyter.Notebook.-.parallel.False.pdf)


Maybe we should confirm the library's version in Jupyter Notebook's environment first.
Here is my information:
numba: 0.48.0
numpy: 1.18.5
stitching: 0.5.3
matplotlib: 3.5.3
